### PR TITLE
Voyager v2

### PR DESCRIPTION
**Pull request by benluelo**
# Plugins and Modules

## Changes

- unique errors for connection and client state verification in ibc-solidity connection handshake verification
- more attributes on generated prost types for `cometbft-rpc`
- more canonical types in `unionlabs` for generated prost types
- the great untyping:
  - `beacon-api-client` is now fully generic over chain spec and does not take a generic for chain spec
  - `unionlabs::ics24::Path` is no longer generic, using only concrete types
  - `unionlabs::ics24::IbcPath` is no longer generic, using only concrete types
    - IbcPath::Value for client and consensus state paths are now just opaque bytes
  - `unionlabs::event::IbcEvent` is no longer generic, using only concrete types
  - all of the IBC `Msg*` are no longer generic, using only concrete types
  - added `Unbounded*` versions of all of the types in `unionlabs::ethereum::beacon` that are generic over ChainSpec, for use in `beacon-api-client`
    - open to better naming here
  - `unionlabs::QueryHeight` is no longer generic, using only the concrete type `Height`
- removed a bunch of old commented out code
- queue_msg:
  - removed `OpEvent`/`QueueMessage::Event` (no longer useful with the new architecture)
- added new `RawAny` type, since some cosmos types use a dynamic list of `Any` values and as such they must be type-erased
- cleaned up `unionlabs::ethereum::beacon` to follow the structure of the rest of the crate (one module/file per type)
- `#[model]` now derives `Eq`

## Closes

- https://github.com/unionlabs/union/issues/1813 (no longer relevant)

## Non-code refactors

- normalize comments in rust code (mostly affected generated code)

---

`chain-utils` is a bit of a mess right now, this crate will be drastically refactored in the coming weeks. i recommend just kinda not looking at it too hard

### CODE REVIEW

File: cosmwasm/ucs01-relay/src/protocol.rs [hunk 1]
```diff
@@ -1127,6 +1127,7 @@ impl<'a> TransferProtocol for Ucs01Protocol<'a> {
 }
 
 #[cfg(test)]
+#[allow(deprecated)] // TODO: Remove usage of mock_info
```
Comment (aeryz) [inline cosmwasm/ucs01-relay/src/protocol.rs:1130]: probably a bad rebase

File: lib/aptos-rpc/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#[derive(Debug, Clone)]
```
Comment (aeryz) [inline lib/aptos-rpc/src/lib.rs:1]: remove

File: lib/cometbft-rpc/src/lib.rs [hunk 3]
```diff
@@ -308,7 +329,7 @@ where
 //             )
 //             .await;
 
-//         dbg!(result);
+//         bg!(result);
```
Comment (aeryz) [inline lib/cometbft-rpc/src/lib.rs:332]: nice diff

File: lib/cometbft-rpc/src/types.rs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    #[serde(deserialize_with = "serde_as::<_, protos::tendermint::p2p::DefaultNodeInfo, _>")]
+    pub node_info: DefaultNodeInfo,
+    pub sync_info: SyncInfo,
+    pub validator_info: ValidatorInfo,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct SyncInfo {
+    pub(crate) catching_up: bool,
```
Comment (aeryz) [inline lib/cometbft-rpc/src/types.rs:49]: make it pub only

File: lib/queue-msg/src/lib.rs [hunk 5]
```diff
@@ -950,93 +847,89 @@ pub fn now() -> u64 {
...
-                    Ok(Some(Op::Data(d))) => {
-                        let data_output = d.clone().handle(&s).unwrap();
-
-                        // run to a fixed point
-                        if data_output != data(d.clone()) {
-                            (None, Ok(vec![data_output]))
-                        } else {
-                            (Some(d), Ok(vec![]))
+    pub(crate) fn step<'b>(
+        &'b self,
```
Comment (aeryz) [inline lib/queue-msg/src/lib.rs:880]: lets put these before the tests

File: lib/voyager-message/src/call.rs [hunk 6]
```diff
@@ -0,0 +1,1438 @@
...
+                    proof_ack: encoded_proof_ack,
+                    proof_height: origin_chain_proof_height,
+                })))
+            }
+
+            Call::MakeMsgRecvPacket(msg) => make_msg_recv_packet(ctx, msg).await,
+
+            Call::MakeMsgAcknowledgement(msg) => make_msg_acknowledgement(ctx, msg).await,
+
+            // Fetch::UpdateHeaders(FetchUpdateHeaders {
```
Comment (aeryz) [inline lib/voyager-message/src/call.rs:717]: remove

File: move/mock_ibc_module.move [hunk 7]
```diff
@@ -0,0 +1,97 @@
+module IBCModuleAddr::IBCModule {
+    use std::vector;
```
Comment (aeryz) [inline move/mock_ibc_module.move:2]: remove this file

File: move/move-ibc/.movement/config.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+profiles:
```
Comment (aeryz) [inline move/move-ibc/.movement/config.yaml:1]: delete

**Review by aeryz — APPROVED** [review]

**Review by cor — APPROVED** [review]

**Pull request merged**